starter-code/scripts/models/article.js

```diff
@@ -40,23 +40,33 @@ Article.loadAll = function(inputData) {
 };
 
 /* This function below will retrieve the data from either a local or remote
- source, process it, then hand off control to the View: */
+source, process it, then hand off control to the View:   $.getJSON()     */
 Article.fetchAll = function() {
   if (localStorage.blogArticles) {
     /* When our data is already in localStorage:
     1. We can process and load it,
     2. Then we can render the index page.  */
+    var parsedData = JSON.parse(localStorage.blogArticles);
+    Article.loadAll(parsedData);
+    articleView.renderIndexPage();
   } else {
     /* Without our localStorage in memory, we need to:
     1. Retrieve our JSON file with $.getJSON
-      1.a Load our json data
-      1.b Store that data in localStorage so that we can skip the server call next time,
-      1.c And then render the index page.*/
+    1.a Load our json data
+    1.b Store that data in localStorage so that we can skip the server call next time,
+    1.c And then render the index page.*/
+
+    $.getJSON('../data/blogArticles.json', function(blogData) {
+      var stringifiedData = JSON.stringify(blogData);
```

Comment:
Nice! If we don't have any data in localStorage, we request it by asking the server.